### PR TITLE
Support pipeline metrics

### PR DESCRIPTION
**Pull request by StefanoFioravanzo**
This PR add the following functionalities:

- Parse `pipeline-metrics` tag
- Parse and validate code blocks belonging to `pipeline-metrics` tag
- Generate code to run pipeline with pipeline metrics
- RPC to let the UI get the current pipeline metrics

### CODE REVIEW

File: kale/nbparser/parser.py [hunk 1]
```diff
@@ -105,42 +110,99 @@ def merge_code(nb_graph, dst, code):
...
+    source = ''
+
+    language = _TAGS_LANGUAGE[:]
+    language.remove(search_tag)
+
     for c in notebook.cells:
         # parse only source code cells
         if c.cell_type != "code":
             continue
         # in case the previous cell was a pipeline-parameter cell and this
```
Comment (elikatsis) [inline kale/nbparser/parser.py]: nit: Fix the comment. It refers to `pipeline-parameter` cell tag

File: kale/utils/utils.py [hunk 2]
```diff
@@ -11,3 +12,27 @@ def random_string(size=5, chars=string.ascii_lowercase + string.digits):
...
+    Pipeline metrics names are Python names that need to conform to KFP's
+    metrics names.
+
+    - Python identifiers: (letter|"_") (letter | digit | "_")*
+    - KFP metric regex: ^[a-z]([-a-z0-9]{0,62}[a-z0-9])?$
+    """
+    mtr = re.sub("_", "-", name)
+    if not validate_kfp_metric_name(mtr):
+        raise ValueError("Pipeline metric {} has an invalid name."
+                         " Valid metric names must be 62 chars long,"
```
Comment (elikatsis) [inline kale/utils/utils.py]:
```suggestion
                         " Valid metric names must be 64 chars long,"
```

File: kale/tests/unit_tests/test_ast.py [hunk 3]
```diff
@@ -119,4 +118,31 @@ def test_parse_assignments_expressions(code, target):
...
+
+
+@pytest.mark.parametrize("code", [
+    'print(foo())',
+    'def',
+    'variable',
+    'print(var)\nname',
+    '_variable'
+])
+def test_parse_assignments_expressions_exc(code):
```
Comment (elikatsis) [inline kale/tests/unit_tests/test_ast.py]:
```suggestion
def test_parse_metrics_expressions_exc(code):
```

**Comment by StefanoFioravanzo** [comment]
@elikatsis I addressed your comments, thank for the nits :) Waiting for your approval

**Pull request merged**